guild_scheduled_event.go

```diff
@@ -16,3 +25,282 @@ type GuildScheduledEvent struct {
+	EntityMetadata     *ScheduledEventEntityMetadata    `json:"entity_metadata"`
+	Name               *string                          `json:"name,omitempty"`
+	PrivacyLevel       *GuildScheduledEventPrivacyLevel `json:"privacy_level"`
+	ScheduledStartTime *Time                            `json:"scheduled_start_time,omitempty"`
+	ScheduledEndTime   *Time                            `json:"scheduled_end_time,omitempty"`
+	Description        *string                          `json:"description,omitempty"`
+	EntityType         *GuildScheduledEventEntityTypes  `json:"entity_type"`
+	Status             *GuildScheduledEventStatus       `json:"status"`
+
+	AuditLogReason string `json:"-"`
+}
+
+func (cse *UpdateScheduledEvent) validate() error {
+	if *cse.EntityType == GuildScheduledEventEntityTypesExternal {
+		if cse.EntityMetadata != nil && cse.EntityMetadata.Location == "" {
+			return ErrMissingScheduledEventLocation
+		}
+
+		if cse.ScheduledEndTime == nil {
+			return ErrMissingScheduledEventEndTime
+		}
+
+		cse.ChannelID = nil
+	}
+
+	if l := len(*cse.Name); !(2 <= l && l <= 100) {
+		return fmt.Errorf("scheduled event name must be 2 or more characters and no more than 100 characters: %w", IllegalValueErr)
+	}
+
+	if *cse.PrivacyLevel != GuildScheduledEventPrivacyLevelGuildOnly {
+		return ErrIllegalScheduledEventPrivacyLevelValue
+	}
+
+	return nil
+}
+
+func (gse *guildScheduledEventQueryBuilder) Update(ID Snowflake, params *UpdateScheduledEvent) (*GuildScheduledEvent, error) {
+	if params == nil {
+		return nil, ErrMissingRESTParams
+	}
+
+	if err := params.validate(); err != nil {
+		return nil, err
+	}
+
+	r := gse.client.newRESTRequest(&httd.Request{
+		Method:      http.MethodPatch,
+		Ctx:         gse.ctx,
+		Endpoint:    endpoint.ScheduledEvent(gse.gid, ID),
+		Body:        params,
+		ContentType: httd.ContentTypeJSON,
+		Reason:      params.AuditLogReason,
+	}, gse.flags)
+	r.factory = func() interface{} {
+		return &GuildScheduledEvent{}
+	}
+
+	return getScheduledEvent(r.Execute)
+}
+
+type GetScheduledEventMembers struct {
+	Limit      uint32    `urlparam:"limit,omitempty"`
+	WithMember bool      `urlparam:"with_member,omitempty"`
+	Before     Snowflake `urlparam:"before,omitempty"`
+	After      Snowflake `urlparam:"after,omitempty"`
+}
+
+type GuildScheduledEventUsers struct {
+	GuildScheduledEventID Snowflake `json:"guild_scheduled_event_id"`
+	User                  `json:"user"`
+	Member                `json:"member"`
+}
+
+// Deprecated: use GetScheduledEventMembers instead
+func (gse *guildScheduledEventQueryBuilder) GetMembers(eventID Snowflake, params *GetScheduledEventMembers) ([]*GuildScheduledEventUsers, error) {
+	const QueryLimit uint32 = 100
+
+	if params == nil {
+		params = &GetScheduledEventMembers{
+			Limit: QueryLimit,
+		}
+	}
+
+	if params.Limit == 0 || params.Limit > QueryLimit {
+		params.Limit = QueryLimit
+	} else {
+		params.Limit = params.Limit
+	}
+
+	panic("TODO: implement this")
```

Comment:
I wonder, is this deprecated by the discord api?

if it is, then the function body can be replaced by the `panic("deprecated by discord")`, just to make life a little easier

Reply:
changed with used method for getMembers